packages/core/src/api/interactions.ts

```diff
@@ -0,0 +1,181 @@
+import type { Snowflake } from 'discord-api-types/v10';
+import {
+	InteractionResponseType,
+	Routes,
+	type APICommandAutocompleteInteractionResponseCallbackData,
+	type APIInteractionResponseCallbackData,
+	type APIModalInteractionResponseCallbackData,
+	type RESTGetAPIWebhookWithTokenMessageResult,
+} from 'discord-api-types/v10';
+import type { WebhooksAPI } from './webhook.js';
+
+export class InteractionsAPI {
+	public constructor(private readonly rest: REST, private readonly webhooks: WebhooksAPI) {}
+
+	/**
+	 * Replies to an interaction
+	 *
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param data - The data to use when replying
+	 */
+	public async reply(
+		interactionId: Snowflake,
+		interactionToken: string,
+		{ files, ...data }: APIInteractionResponseCallbackData & { files?: RawFile[] },
+	) {
+		await this.rest.post(Routes.interactionCallback(interactionId, interactionToken), {
+			files,
+			body: {
+				type: InteractionResponseType.ChannelMessageWithSource,
+				data,
+			},
+		});
+	}
+
+	/**
+	 * Defers the reply to an interaction
+	 *
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 */
+	public async defer(interactionId: Snowflake, interactionToken: string) {
+		await this.rest.post(Routes.interactionCallback(interactionId, interactionToken), {
+			body: {
+				type: InteractionResponseType.DeferredChannelMessageWithSource,
+			},
+		});
+	}
+
+	/**
+	 * Defers an update from a message component interaction
+	 *
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 */
+	public async deferMessageUpdate(interactionId: Snowflake, interactionToken: string) {
+		await this.rest.post(Routes.interactionCallback(interactionId, interactionToken), {
+			body: {
+				type: InteractionResponseType.DeferredMessageUpdate,
+			},
+		});
+	}
+
+	/**
+	 * Reply to a deferred interaction
+	 *
+	 * @param applicationId - The application id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param data - The data to use when replying
+	 */
+	public async followUp(
+		applicationId: Snowflake,
+		interactionToken: string,
+		data: APIInteractionResponseCallbackData & { files?: RawFile[] },
+	) {
+		await this.webhooks.execute(applicationId, interactionToken, data);
+	}
+
+	/**
+	 * Edits the initial reply to an interaction
+	 *
+	 * @param applicationId - The application id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param data - The data to use when editing the reply
+	 * @param messageId - The id of the message to edit, if omitted the original reply will be edited
+	 */
+	public async editReply(
+		applicationId: Snowflake,
+		interactionToken: string,
+		data: APIInteractionResponseCallbackData & { files?: RawFile[] },
+		messageId?: string,
```

Comment:
Won't block the pr for this but i don't like how the message id is at the end... then again outside of making it an object for the params idk a better solution

Reply:
Making this an object seems fine?